python/GafferUI/ExpressionUI.py

```diff
@@ -181,6 +181,31 @@ def expressionContextMenuSignal( cls ) :
 
 		return cls.__expressionContextMenuSignal
 
+	__highlighters = {}
+	## Registers a function to return a `CodeWidget.Highlighter`
+	# for a particular engine type. `highlighter` will be passed
+	# an Expression node and must return a Completer.
+	@classmethod
+	def registerHighlighter( cls, language, highlighter ) :
+
+		cls.__highlighters[language] = highlighter
+
+	__completers = {}
+	## Registers a function to return a `CodeWidget.Completer`
+	# for a particular engine type. `completer` will be passed
+	# an Expression node and must return a Completer.
+	@classmethod
+	def registerCompleter( cls, language, completer ) :
+
+		cls.__completers[language] = completer
+
+	__commentPrefixes = {}
+	## Registers the comment prefix for a particular engine type.
+	@classmethod
+	def registerCommentPrefix( cls, language, commentPrefix ) :
+
+		cls.__commentPrefixes[language] = commentPrefix
+
```

Comment:
Is there any good reason to have 3 separate registries here, instead of one that holds the highlighter, completer, and commentPrefix together?

Reply:
No, but I can't think of a compelling reason to join them either.